api/submission/fetch_submission_history.go

```diff
@@ -0,0 +1,49 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+    "github.com/eriq-augustine/autograder/model"
+)
+
+type FetchSubmissionHistoryRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser       core.TargetUserSelfOrGrader `json:"target-email"`
+}
+
+type FetchSubmissionHistoryResponse struct {
+    FoundUser        bool `json:"found-user"`
+    FoundSubmissions bool `json:"found-submissions"`
+    GradingResults []*model.GradingResult `json:"grading-results"`
+}
+
+func HandleFetchSubmissionHistory(request *FetchSubmissionHistoryRequest) (*FetchSubmissionHistoryResponse, *core.APIError) {
+    response := FetchSubmissionHistoryResponse{
+        FoundUser:        false,
+        FoundSubmissions: false,
+        GradingResults:   make([]*model.GradingResult, 0),
+    }
+
+    if !request.TargetUser.Found {
+        return &response, nil;
+    }
+
+    response.FoundUser = true;
+
+    gradingResults, err := db.GetSubmissionResultHistory(request.Assignment, request.TargetUser.Email);
+    if err != nil {
+        return nil, core.NewInternalError("-604", &request.APIRequestCourseUserContext, "Failed to get submissions.").
```

Comment:
Wrong locator.

Be real careful here, either this was just a slip (which happens), or you didn't fully understand how the locators worked and guessed.
For something like this it can be hard to catch if missed, so make sure to ask.
Every locator should be unique so we can trace exactly where an error came from (e.g. a student has an error, and we only need this number to figure out what happened).
Each package gets allocated 100 numbers, choose the next available one for this package.

Reply:
I committed a fix to this.  I did not fully understand, but I do now.  Thank you.  I will make sure to ask in the future.